### PR TITLE
beef up dogfood application

### PR DESCRIPTION
**Pull request by takakonishimura**
## What does this PR do?

- [ ] Adds new functionality
- [ ] Alters existing functionality
- [ ] Fixes a bug
- [x] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- This is a cosmetic PR to make the dogfood application more realistic (you can request more than one kind of food)
- While the original dogfood application is sufficient to test the gRPC disruption which operates as the end point level, a more realistic use case is (1) cuter (2) more intuitive for new developers to understand nuances and (3) allows testing of future where we will mock results based on request content

## Code Quality Checklist

- [x] The documentation is up to date.
- [x] My code is sufficiently commented and passes continuous integration checks.
- [x] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [ ] I leveraged continuous integration testing
    - [ ] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [x] I manually tested the following steps:
    - See `docs/grpc_disruption/setup_instructions.md`
    - [x] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

File: grpcdogfood/server/dogfood_server.go [hunk 1]
```diff
@@ -18,18 +19,47 @@ import (
...
+
+	fmt.Printf("| * DECLINED ORDER - %s\n", req.String())
+
+	return nil, errors.New("Sorry, we don't deliver food for your " + req.Animal + " =(")
 }
 
 func (s *chaosDogfoodServer) GetCatalog(ctx context.Context, req *emptypb.Empty) (*pb.CatalogReply, error) {
-	fmt.Println("* catalog delivered")
-	return &pb.CatalogReply{Items: []*pb.CatalogItem{}}, nil
+	fmt.Println("| returned catalog")
```
Comment (takakonishimura) [inline grpcdogfood/server/dogfood_server.go:51]: returns more animals

File: grpcdogfood/server/dogfood_server.go [hunk 2]
```diff
@@ -18,18 +19,47 @@ import (
 
 const serverAddr = "localhost:50051"
 
+var catalog = map[string]string{
```
Comment (takakonishimura) [inline grpcdogfood/server/dogfood_server.go:22]: single variable holding catalog details

File: docs/grpc_disruption/setup_instructions.md [hunk 3]
```diff
@@ -16,28 +16,37 @@ To locally run the dogfood client, call `go run grpcdogfood/client/dogfood_clien
...
 connecting to localhost:50051...
-| got catalog: 0 items returned
-| ordered: Mock Reply
-| got catalog: 0 items returned
-| ordered: Mock Reply
-| got catalog: 0 items returned
-| ordered: Mock Reply
-| got catalog: 0 items returned
-| ordered: Mock Reply
+| catalog: 3 items returned (cat, dog, cow)
```
Comment (takakonishimura) [inline docs/grpc_disruption/setup_instructions.md:19]: This diff tells you everything important that changed in the client file

File: docs/grpc_disruption/setup_instructions.md [hunk 4]
```diff
@@ -16,28 +16,37 @@ To locally run the dogfood client, call `go run grpcdogfood/client/dogfood_clien
...
 listening on localhost:50051...
-* catalog delivered
-* cat food ordered
-* catalog delivered
-* cat food ordered
-* catalog delivered
-* cat food ordered
-* catalog delivered
-* cat food ordered
+| returned catalog
```
Comment (takakonishimura) [inline docs/grpc_disruption/setup_instructions.md:38]: This diff tells you everything important that changed in the server file

**Review by takakonishimura — COMMENTED** [review]
helper comments for reviewer

**Review by Devatoria — APPROVED** [review]

**Pull request merged**